core/dbt/exceptions.py

```diff
@@ -2132,12 +2132,25 @@ def __init__(self, yaml_columns, sql_columns):
 
     def get_message(self) -> str:
         msg = (
-            "Contracts are enabled for this model. "
+            "This model has an enforced contract that failed.\n"
             "Please ensure the name, data_type, and number of columns in your `yml` file "
             "match the columns in your SQL file.\n"
-            f"Schema File Columns: {self.yaml_columns}\n"
-            f"SQL File Columns: {self.sql_columns}"
+            # separate column headers by 20 spaces
+            f"{'yaml col:' : <20} sql col:\n"
         )
+
+        # print columns pretty
+        # split yaml and sql strings into list
+        yaml_list = self.yaml_columns.split(', ')
+        sql_list = self.sql_columns.split(', ')
```

Comment:
Ah - I didn't realize that these are the stringified versions of `yaml_columns` and `sql_columns`, but that is indeed exactly how they're being passed in here (`string_yaml_columns` and `string_sql_columns`):

https://github.com/dbt-labs/dbt-core/blob/73ff49720021334bd793e55d9d1a198a85ccef5d/core/dbt/include/global_project/macros/materializations/models/table/columns_spec_ddl.sql#L45-L65

It is also true that the logic we're using to compare columns one-by-one, and the logic we'd want to use to display pretty output, should be quite similar.

So, a couple of potential approaches here:
1. Pass the dict (not str) representation of these columns into the Python exception, so that we can do the same sort of comparison over here again
2. Since the Jinja macro is already detecting differences - pass only the _ordered_ set of _disagreeing_ columns, from the `assert_columns_equivalent` macro, into this exception. (But what about the case where the two column lengths differ? We'd still want to detect which columns are present in one, and missing from the other)
3. Move this comparison logic into a Python method, and have `assert_columns_equivalent` macro call that method instead of writing this logic in Jinja. That method would call this exception if it detects differences.

(1) is probably simplest with the existing implementation - just switch out `string_yaml_columns` + `string_sql_columns` for `yaml_columns` and `sql_columns` in the macro definition linked above.

@MichelleArk Curious if you have an opinion on what the right way to go here would be!

Reply:
@jtcohen6 Alright, what I just pushed should work. It's not my favorite solution because the columns are actually passed as a list of dictionaries `[{col1 : col1 INT}, ...]`. I could've removed the list to avoid using so many for loops, but decided mimic the exception call in the .sql file as best I could. Cuz if it aint broke don't fix it. Attached is an image of how the error prints!
<img width="642" alt="Screenshot 2023-04-02 at 6 42 59 PM" src="https://user-images.githubusercontent.com/80537707/229396460-7b1f93d7-13d4-4a50-8662-411e4a8bafb0.png">

As for the rebase issue, if you're happy with these changes I'll open a new PR with the modified files.

Reply:
@jtcohen6 Alright, what I just pushed should work. It's not my favorite solution because the columns are actually passed as a list of dictionaries `[{col1 : col1 INT}, ...]`. I could've removed the list to avoid using so many for loops, but decided mimic the exception call in the .sql file as best I could. Cuz if it aint broke don't fix it. Attached is an image of how the error prints!
<img width="642" alt="Screenshot 2023-04-02 at 6 42 59 PM" src="https://user-images.githubusercontent.com/80537707/229396460-7b1f93d7-13d4-4a50-8662-411e4a8bafb0.png">

As for the rebase issue, if you're happy with these changes I'll open a new PR with the modified files.

Reply:
@kentkr This is a big improvement!

Sorry for the delay getting back to you. To have a better chance of getting this over the finish line, I'm going to open a new PR with
- rebased commits to fix the CLA issue
- a few more aesthetic touch-ups (e.g. highlighting columns with mismatches, and using our existing library for printing markdown tables to stdout)